str/catalog_design/polymorphic_site_extractor.py

```diff
@@ -0,0 +1,157 @@
+    filtered_mt = filtered_mt.filter_rows(hl.len(filtered_mt.alleles) > 1)
+
+    # at a biallelic locus, remove loci that are monomorphic for 1 ALT allele
+    filtered_mt = filtered_mt.filter_rows(
+        ~(
+            (hl.len(filtered_mt.variant_qc.AC) == 2)
+            & (filtered_mt.variant_qc.AC[0] == 0)
+        )
+    )
+
+    # collect the REPIDs into one list
+    rep_id_list = filtered_mt.info.REPID.collect()
+
+    return rep_id_list
+
+
+def catalog_filter(rep_id_list, catalog_path, gcs_output_path):
+    """Retains loci in a JSON file that intersect with a list of REPIDs (STR equiv. of rsids) and writes the filtered catalog to a new JSON file"""
+    with to_path(catalog_path).open('r') as json_file:
+        # Load the JSON content
+        catalog = json.load(json_file)
+
+    filtered_data = []
+
+    # Make the rep_id_list into a set
+    polymorphic_rep_id_set = set(rep_id_list)
+
+    for entry in catalog:
+        # Check if 'VariantId' exists, use 'LocusId' otherwise
+        entry_variant_ids = (
+            set(entry['VariantId']) if 'VariantId' in entry else {entry['LocusId']}
+        )
+
+        # Append to filtered_data if the entry Variant Id intersects with polymorphic_rep_id_set
+        if entry_variant_ids & polymorphic_rep_id_set:
+            filtered_data.append(entry)
+
+    # Write to output
+    with to_path(gcs_output_path).open('w') as out_file:
+        json.dump(filtered_data, out_file, indent=2)
+    return filtered_data
+
+
+def catalog_sharder(filtered_data, chunk_size, folder_name):
+    """Shards a filtered catalog JSON file into chunks of size chunk_size"""
+    if not isinstance(filtered_data, list):
+        print('Invalid JSON format. The file should contain a list.')
+        return
+
+    total_entries = len(filtered_data)
+    num_chunks = total_entries // chunk_size
+    remainder = total_entries % chunk_size
+
+    if remainder > 0:
+        num_chunks += 1
+
+    for i in range(num_chunks):
+        start_idx = i * chunk_size
+        end_idx = (i + 1) * chunk_size
+        chunk_data = filtered_data[start_idx:end_idx]
+
+        output_file_path = output_path(f'{folder_name}/chunk_{i + 1}.json', 'analysis')
+
+        with to_path(output_file_path).open('w') as output_file:
+            json.dump(chunk_data, output_file, indent=2)
+
+        print(f'Chunk {i + 1} created: {output_file_path}')
+
+    print(f'Total entries: {total_entries}')
+    print(f'{num_chunks} chunks written successfully!')
+
+
+@click.option(
+    '--vcf-path',
+    help='GCS file path VCF',
+    type=str,
+)
+@click.option(
+    '--catalog-path',
+    help='GCS file path to catalog JSON to be filtered',
+    type=str,
+)
+@click.option(
+    '--chunk-size',
+    help='Number of entries per shard',
+    type=int,
+    default=100000,
+)
+@click.option(
+    '--folder-name',
+    help='Name of output folder to store shards',
+    type=str,
+    default='sharded_polymorphic_catalog',
+)
```

Comment:
```suggestion
@click.option('--vcf-path', help='GCS file path VCF')
@click.option('--catalog-path', help='GCS file path to catalog JSON to be filtered')
@click.option(
    '--chunk-size',
    help='Number of entries per shard',
    type=int,
    default=100000,
)
@click.option('--folder-name', help='Name of output folder to store shards', default='sharded_polymorphic_catalog')
```

This is just being really pedantic - str is the default type for click arguments so you won't need to specify it